File: fds-backend/user-management-service/src/main/java/com/yokke/usermanagement/user_account_rule/UserAccountRuleDto.java
package com.yokke.usermanagement.user_account_rule;

import com.yokke.base.dto.BaseDto;
import com.yokke.usermanagement.rule.RuleDto;
import com.yokke.usermanagement.user_account.UserAccountDTO;
import lombok.*;

import java.io.Serializable;
import java.time.OffsetDateTime;

/**
 * DTO for {@link UserAccountRule}
 */
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class UserAccountRuleDto extends BaseDto implements Serializable {
    String userRuleId;
    Boolean isActive;
    Integer priority;
    UserAccountDTO userAccount;
    RuleDto rule;



}

File: fds-dashboard/src/packages/user-management-service/src/user-account-rule/dto/user-account-rule.dto.ts
import type { BasePageableDto } from "@/packages/base/src/dto/base-pageable.dto";
import type { BaseDto } from "@/packages/base/src/dto/base.dto";

import type { RuleDto } from "@/packages/user-management-service/src/rule/dto/rule.dto";
import type { UserAccountResponse } from "@/packages/user-management-service/src/user-account/response/user-account.response";

export type UserAccountRuleDto = {
  userRuleId: string;
  isActive: boolean;
  priority: number;
  userAccount: UserAccountResponse;
  rule: RuleDto;
} & BaseDto;

export type UserAccountRulePageableDto = {
  content: UserAccountRuleDto[];
} & BasePageableDto;
